### PR TITLE
Improve handling for documentation and other private links

### PR DESCRIPTION
**Pull request by razvanpapadopol**
Fixes https://github.com/Automattic/abacus/issues/801 
Fixes https://github.com/Automattic/abacus/issues/803

* Add <PrivateLink /> component that wraps the Material UI `<Link />`.
* Limit rendering the private links depending on the environment (only in production) when the app is built using `mockConfig`.
* Switch to using shortlinks for private resources.

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)

### CODE REVIEW

**Review by aaronyan — APPROVED** [review]
LGTM! Thanks for adding the new `PrivateLink` component, adding new tests and adjusting the existing ones.

File: package.json [hunk 1]
```diff
@@ -97,6 +97,7 @@
     "isomorphic-fetch": "^3.0.0",
     "jest": "26.6.0",
     "jest-puppeteer": "^6.1.0",
+    "jest-watch-typeahead": "^0.6.5",
```
Comment (razvanpapadopol) [inline package.json:100]: After the [node update](https://github.com/Automattic/abacus/pull/806), a jest watch plugin seems to be needed in order to update snapshots. I'll check this again when [updating packages](https://github.com/Automattic/abacus/pull/744).

**Comment by jessie-ross** [comment]
There is nothing wrong with the code here, and this isn't a detrimental change so feel free to merge - using the short-links is good in fact.

But I think the `PrivateLink` functionality is a bit of a security micro-optimisation - the links are still present in the code and if we want to stop scraping we could use a robots.txt.

File: package.json (hunk 1 above)
Comment (jessie-ross) [reply package.json:100]: :(

**Comment by razvanpapadopol** [comment]
> There is nothing wrong with the code here, and this isn't a detrimental change so feel free to merge - using the short-links is good in fact.

Thanks for having a look! 

> But I think the `PrivateLink` functionality is a bit of a security micro-optimisation - the links are still present in the code and if we want to stop scraping we could use a robots.txt.

Oh, this is not for security but to have a clean open source / Demo version of Abacus (https://github.com/Automattic/abacus/issues/801). Anyway, I don't think that usual crawlers will index the links if they are not accessible from the UI 🤷🏼

**Pull request merged**